cranelift/filetests/src/test_wasm.rs

```diff
@@ -1,198 +0,0 @@
-//! Test runner for `.wat` files to exercise CLIF-to-Wasm translations.
-
-mod config;
-mod env;
-
-use anyhow::{bail, ensure, Context, Result};
-use config::TestConfig;
-use cranelift_codegen::ir::Function;
-use cranelift_codegen::isa::TargetIsa;
-use cranelift_control::ControlPlane;
-use env::ModuleEnv;
-use serde::de::DeserializeOwned;
-use serde_derive::Deserialize;
-use similar::TextDiff;
-use std::{fmt::Write, path::Path};
-
-/// Run one `.wat` test.
-pub fn run(path: &Path, wat: &str) -> Result<()> {
-    debug_assert_eq!(path.extension().unwrap_or_default(), "wat");
-
-    // The test config source is the leading lines of the WAT file that are
-    // prefixed with `;;!`.
-    let config_lines: Vec<_> = wat
-        .lines()
-        .take_while(|l| l.starts_with(";;!"))
-        .map(|l| &l[3..])
-        .collect();
-    let config_text = config_lines.join("\n");
-
-    let config: TestConfig =
-        toml::from_str(&config_text).context("failed to parse the test configuration")?;
-    log::debug!("Wasm test config = {config:#?}");
-
-    config
-        .validate()
-        .context("test configuration is malformed")?;
-
-    let parsed = cranelift_reader::parse_sets_and_triple(&config.settings, &config.target)
-        .context("invalid ISA target or Cranelift settings")?;
-    let flags_or_isa = parsed.as_fisa();
-    ensure!(
-        flags_or_isa.isa.is_some(),
-        "Running `.wat` tests requires specifying an ISA"
-    );
-    let isa = flags_or_isa.isa.unwrap();
-
-    let mut env = ModuleEnv::new(isa, config.clone());
-
-    let wasm = wat::parse_str(wat).context("failed to parse the test WAT")?;
-    let mut validator = wasmparser::Validator::new_with_features(
-        cranelift_wasm::ModuleEnvironment::wasm_features(&env),
```

Comment:
Can the cranelift-wasm and wasmparser dependencies be removed from filetests?

Reply:
Oh indeed they can!